debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/OutboxEventRouterIT.java

```diff
@@ -205,5 +228,126 @@ public void shouldSupportAllFeatures() throws Exception {
+        outboxEventRouter = new EventRouter<>();
+        final Map<String, String> config = new HashMap<>();
+        config.put("table.field.event.schema.version", "version");
+        config.put("table.field.event.timestamp", "createdat");
+        config.put("route.tombstone.on.empty.payload", "true");
+        config.put(
+                "table.fields.additional.placement",
+                "version:envelope:eventVersion," +
+                        "aggregatetype:envelope:aggregateType," +
+                        "somebooltype:envelope:someBoolType," +
+                        "somebooltype:header," +
+                        "is_deleted:envelope:deleted"
+        );
+        outboxEventRouter.configure(config);
+
+        TestHelper.execute("ALTER TABLE outboxsmtit.outbox add version int not null;");
+        TestHelper.execute("ALTER TABLE outboxsmtit.outbox add somebooltype boolean not null;");
+        TestHelper.execute("ALTER TABLE outboxsmtit.outbox add createdat timestamp without time zone not null;");
+        TestHelper.execute("ALTER TABLE outboxsmtit.outbox add is_deleted boolean not null default false;");
+
+        TestHelper.execute(createEventInsert(
+                UUID.fromString("a9d76f78-bda6-48d3-97ed-13a146163218"),
+                "UserUpdated",
+                "UserEmail",
+                "a9d76f78",
+                "{\"email\": \"brandon@bbrownsound.com\"}",
+                ", 1, true, TIMESTAMP '2019-03-24 20:52:59', true"
+        ));
+
+        SourceRecords actualRecords = consumeRecordsByTopic(1);
+        assertThat(actualRecords.topics().size()).isEqualTo(1);
+
+        SourceRecord newEventRecord = actualRecords.recordsForTopic(topicName("outboxsmtit.outbox")).get(0);
+        SourceRecord eventRouted = outboxEventRouter.apply(newEventRecord);
+
+        // Validate metadata
+        assertThat(eventRouted.valueSchema()).isNotNull();
+        assertThat(eventRouted.timestamp()).isEqualTo(1553460779000000L);
+        assertThat(eventRouted.topic()).isEqualTo("outbox.event.useremail");
+
+        // Validate headers
+        Headers headers = eventRouted.headers();
+        assertThat(headers.size()).isEqualTo(2);
+        Header headerId = headers.lastWithName("id");
+        assertThat(headerId.schema()).isEqualTo(Uuid.schema());
+        assertThat(headerId.value()).isEqualTo("a9d76f78-bda6-48d3-97ed-13a146163218");
+        Header headerBool = headers.lastWithName("somebooltype");
+        assertThat(headerBool.schema()).isEqualTo(SchemaBuilder.BOOLEAN_SCHEMA);
+        assertThat(headerBool.value()).isEqualTo(true);
+
+        // Validate Key
+        assertThat(eventRouted.keySchema()).isEqualTo(SchemaBuilder.STRING_SCHEMA);
+        assertThat(eventRouted.key()).isEqualTo("a9d76f78");
+
+        // Validate message body
+        assertThat(eventRouted.value()).isNotNull();
+    }
+
+    @Test
+    public void shouldConsumeTombstoneEvents() throws Exception {
```

Comment:
Would be nice to have a similar test, inserting null, but with the option as false, this way we preserve the feature, right now we only have tests with payload in the whole suite

Reply:
Yeah that seems reasonable!

Reply:
I'm gonna be out with family over the weekend but will have this and the other prs availble monday evening! 🎉